src/bp/ui-studio/src/web/views/OneFlow/diagram/index.tsx

```diff
@@ -1158,11 +1197,19 @@ class Diagram extends Component<Props> {
             >
               <DiagramWidget
                 ref={w => (this.diagramWidget = w)}
+                allowCanvasZoom={false}
```

Comment:
Why remove an intuitive feature? We can keep the zoom toolbar while keeping the mouse scrollwhell.

Reply:
Tried to allow it and listen to the zoom event so I could update the zoomLevel for the buttons on scroll zoom and I couldn't find a way to do it so it would create a really weird behavior

Reply:
You can add a listener to the diagram model.

```js
// in manager.ts
this.activeModel.addListener({ zoomUpdated: e => console.log(e.zoom) })
```

It will also fix another issue, because if you refresh the page while having less space (when the developer panel is opened, for example), it incorrectly displays 100%.

![image](https://user-images.githubusercontent.com/42552874/93217248-cb571d00-f736-11ea-858a-803166ca7df4.png)

Reply:
I don't seem to be able to make redux work in the manager.ts and I adding the listener to `this.diagramEngine.diagramModel` in the diagram file doesn't get triggered... I'll just allow the scroll zoom and have buttons that will zoom starting from 100%, might be a bit weird, but I've spent way too much time on this shit already

Reply:
Here's the complete solution:
```js
// manager.ts
this.activeModel.addListener({ zoomUpdated: e => this.storeDispatch.zoomToLevel(e.zoom) })

// oneflow/diagram/index.ts - update the manager initializer
this.manager = new DiagramManager(this.diagramEngine, {
  switchFlowNode: this.props.switchFlowNode,
  zoomToLevel: this.props.zoomToLevel
})
```